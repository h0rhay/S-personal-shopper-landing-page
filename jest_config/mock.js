export const mockTheme = {
  themeType: 'dark',
  breakpoints: {
    small: '(max-width: 812px)',
    medium: '(min-width: 813px)',
    large: '(min-width: 960px)',
    deviceSwitchSm: '(max-width: 896px)',
    deviceSwitchLg: '(min-width: 897px)',
  },
  typography: {
    mobile: {
      headerOne: {
        fontSize: '3rem',
      },
      headerTwo: {
        fontSize: '1.75rem',
      },
      displayHeading: {
        fontSize: 'calc(1.5rem * 2)',
      },
      itemHeading: {
        fontSize: '1.125rem',
      },
    },
    desktop: {
      headerOne: {
        fontSize: '4rem',
      },
      headerTwo: {
        fontSize: '2rem',
      },
      displayHeading: {
        fontSize: 'calc(4rem * 2)',
      },
      itemHeading: {
        fontSize: '1.125rem',
      },
    },
  },
  text: '#ffffff',
  primary: '#212121',
  primaryAccent: '#2E2E2E',
  accent: '#ffe256',
  shadow: 'rgba(0, 0, 0, 0.4)',
};
