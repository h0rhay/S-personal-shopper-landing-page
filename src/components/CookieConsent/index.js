import React, { useState, useEffect } from 'react';
import { useLocation } from '@reach/router';
import { initializeAndTrack } from 'gatsby-plugin-gdpr-cookies';
import styled, { css } from 'styled-components';
import SiteWidthWrapper from '../SiteWidthWrapper';
import Button from '../Button';

const CookieBlock = styled.section`
  position: absolute;
  ${({ theme }) => css`
    @media ${theme.breakpoints.small} {
      position: fixed;
    }
  `}
  width: 100%;
  background: #ffe256;
  z-index: 999;
  bottom: 0;
  padding: 2rem 0;
  * {
    color: #212121;
  }
`;

const CookieHeadingWrap = styled(SiteWidthWrapper)`
  * {
    margin-bottom: 0.5rem;
  }
`;

const CookieWrap = styled(SiteWidthWrapper)`
  display: flex;
  align-items: center;
  .ck-info {
    width: 70%;
  }
  > div > div {
    margin-left: 1rem;
  }
  .mb-msg {
    display: none;
  }
  .dskt-msg {
    display: block;
  }
  a {
    font-family: 'DINNextLTPro-Medium';
    text-transform: none;
  }
  button {
    font-family: 'DINNextLTPro-Medium';
    text-transform: none;
    min-width: 10rem;
    background: #ffe256;
    outline-color: #212121;
    color: #212121;
    &:hover {
      background: #212121;
      outline-color: #ffe256;
      color: #ffe256;
    }
  }
  .accept {
    button {
      outline-color: transparent;
      background: #fff;
      &:hover {
        background: #212121;
        color: #fff;
      }
    }
  }
  ${({ theme }) => css`
    @media ${theme.breakpoints.small} {
      display: block;
      button {
        min-width: 100%;
      }
      p,
      a {
        font-size: 80%;
      }
      .ck-info {
        margin-bottom: 1rem;
        width: 100%;
      }
      .ck-btn {
        width: 50%;
        float: left;
        button {
          margin: 0 auto;
          display: block;
        }
      }
    } ;
  `};
`;

const isBrowser = () => typeof window !== 'undefined';

const getValue = (key, defaultValue) =>
  isBrowser() && window.localStorage.getItem(key) ? JSON.parse(window.localStorage.getItem(key)) : defaultValue;

const setValue = (key, value) => {
  window.localStorage.setItem(key, JSON.stringify(value));
};

const useStickyState = (defaultValue, key) => {
  const [value, setter] = useState(() => getValue(key, defaultValue));

  useEffect(() => {
    setValue(key, value);
  }, [key, value]);

  return [value, setter];
};

const CookieConsent = () => {
  const location = useLocation();
  if (isBrowser()) {
    initializeAndTrack(location);
  }

  const [bannerHidden, setBannerHidden] = useStickyState(false, 'consentCookieHidden');

  const enableAnalytics = () => {
    document.cookie = 'gatsby-gdpr-google-analytics=true';
    setBannerHidden(true);
  };

  const disableAnalytics = () => {
    setBannerHidden(true);
  };

  return (
    // eslint-disable-next-line react/jsx-no-useless-fragment
    <>
      {!bannerHidden && (
        <CookieBlock>
          <CookieHeadingWrap>
            <h3>Giving you the best experience</h3>
          </CookieHeadingWrap>
          <CookieWrap>
            <div className="ck-info">
              <p>
                We use cookies to personalize content and analyze our traffic. These cookies and similar technologies
                help to personalise content and analyse website performance. Learn exactly how we do this in our{' '}
                <a
                  href="https://www.selfridges.com/features/info/our-corporate-policies/privacy-cookie-policy"
                  target="_blank"
                  rel="noopener noreferrer"
                >
                  Cookie Policy
                </a>
              </p>
            </div>
            <div className="ck-btn">
              <Button onClick={disableAnalytics} type="button">
                No thanks
              </Button>
            </div>
            <div className="ck-btn accept">
              <Button onClick={enableAnalytics} type="button">
                Accept
              </Button>
            </div>
          </CookieWrap>
        </CookieBlock>
      )}
    </>
  );
};

export default CookieConsent;
