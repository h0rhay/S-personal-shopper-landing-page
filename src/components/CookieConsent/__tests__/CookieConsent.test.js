import React from 'react';
import { render, screen } from '@testing-library/react';
// import { useLocation } from '@reach/router'; // mocked
// import { initializeAndTrack } from 'gatsby-plugin-gdpr-cookies'; // mocked
import { ThemeProvider } from 'styled-components';
import { mockTheme } from '../../../../jest_config/mock';
import CookieConsent from '../index';

jest.mock('@reach/router');
jest.mock('gatsby-plugin-gdpr-cookies');
jest.mock('gatsby-link', () => <a href="mock-test">Mock Link</a>);

const renderComponent = ({ theme }) =>
  render(
    <ThemeProvider theme={theme}>
      <CookieConsent />
    </ThemeProvider>
  );

afterEach(() => {
  localStorage.removeItem('consentCookieHidden');
});

describe('Cookie consent banner', () => {
  it('should match screenshot', () => {
    const { container } = renderComponent({ theme: mockTheme });
    expect(container).toMatchSnapshot();
  });
});

describe('Cookie consent banner', () => {
  it('should be visible initially', () => {
    renderComponent({ theme: mockTheme });
    expect(screen.getByText(/Giving you the best experience/i)).toBeVisible();
  });
});

describe('Cookie consent banner', () => {
  it('should dissapear when accept button is clicked', () => {
    renderComponent({ theme: mockTheme });
    const banner = screen.getByText(/Giving you the best experience/i);
    screen.getByRole('button', { name: /accept/i }).click();
    expect(banner).not.toBeInTheDocument();
  });
});

describe('Cookie consent banner', () => {
  it('should dissapear when dismiss button is clicked', () => {
    renderComponent({ theme: mockTheme });
    const banner = screen.getByText(/Giving you the best experience/i);
    screen.getByRole('button', { name: /no thanks/i }).click();
    expect(banner).not.toBeInTheDocument();
  });
});
