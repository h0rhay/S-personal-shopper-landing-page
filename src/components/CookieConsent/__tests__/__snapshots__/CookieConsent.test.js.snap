// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cookie consent banner should match screenshot 1`] = `
<div>
  <section
    class="sc-dkPtRN jhSkRV"
  >
    <div
      class="sc-bdvvtL kiJDep sc-hKwDye gPrQbD"
    >
      <h3>
        Giving you the best experience
      </h3>
    </div>
    <div
      class="sc-bdvvtL kiJDep sc-eCImPb fUYyNN"
    >
      <div
        class="ck-info"
      >
        <p>
          We use cookies to personalize content and analyze our traffic. These cookies and similar technologies help to personalise content and analyse website performance. Learn exactly how we do this in our
           
          <a
            href="https://www.selfridges.com/features/info/our-corporate-policies/privacy-cookie-policy"
            rel="noopener noreferrer"
            target="_blank"
          >
            Cookie Policy
          </a>
        </p>
      </div>
      <div
        class="ck-btn"
      >
        <div
          class="sc-gsDKAQ ighmbi"
        >
          <button
            type="button"
          >
            No thanks
          </button>
        </div>
      </div>
      <div
        class="ck-btn accept"
      >
        <div
          class="sc-gsDKAQ ighmbi"
        >
          <button
            type="button"
          >
            Accept
          </button>
        </div>
      </div>
    </div>
  </section>
</div>
`;
